core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -227,13 +231,15 @@ import scala.util.control.NonFatal
   private var delayedPollInFlight = false
 
   def receive: Receive = LoggingReceive {
-    case Assign(tps) =>
+    case Assign(assignedTps) =>
       scheduleFirstPollTask()
-      checkOverlappingRequests("Assign", sender(), tps)
+      checkOverlappingRequests("Assign", sender(), assignedTps)
       val previousAssigned = consumer.assignment()
-      consumer.assign((tps.toSeq ++ previousAssigned.asScala).asJava)
-      val assignedOffsets = tps.map(tp => tp -> consumer.position(tp)).toMap
-      partitionAssigned(assignedOffsets)
+      consumer.assign((assignedTps.toSeq ++ previousAssigned.asScala).asJava)
+      if (commitRefreshing.enabled) {
+        val assignedOffsets = assignedTps.map(tp => tp -> consumer.position(tp, positionTimeout)).toMap
+        commitRefreshing.assignedPositions(assignedTps, assignedOffsets)
+      }
```

Comment:
Instead of checking `commitRefreshing.enabled` we could call `commitRefreshing.assignedPositions` with `assignedTps` and `consumer`, where it would do the right thing, and in the NoOp implementation it would do nothing.